actblue/wordpress_plugin
# Set up deployment scripts for CircleCI

## What this does

Adds scripts for the deployment of plugin assets (for the WordPress plugins page) and the plugin source. This also hooks those scripts up to a circle CI workflow.

## To test

This is how I tested this locally:

1. Make sure that you have subversion installed on your machine. I did this with homebrew on a Mac:

    ```
    brew install subversion
    ```

2. Then create an svn repository somewhere on your machine. I'll do this in my `/tmp/` directory (note that you don't have to _be_ in your home directory to run this command):

    ```
    svnadmin create /tmp/wp-svn-actblue-contributions
    ```

3. Now go back to this git repository in a terminal. Just for testing purposes, we'll need to tag this git repo to mimic the production tagging functionality. Tag the repo with a v1.0.0 tag. Note that the script currently expects the leading `v` in the tag; we can adjust this as needed for ActBlue's tagging workflow, but for now git tags require that leading `v`:

    ```
    git tag v1.0.0
    ```

4. Since we're testing this locally, we need to ensure that the scripts point to the svn repo we made in step 2. Call the following scripts, noting the following things:
    - We're setting the `SVN_URL` environment to point to our local repository before calling the script.
    - We need to pass a username and password to the script. Since we're testing locally these can be dummy strings, but when committing to the WordPress svn repository these will have to reflect the authorized user's username and password. In lieu of passing these values as parameters to the script, you could also locally set environment variables for `WP_ORG_USERNAME` and `WP_ORG_PASSWORD`.

    ```
    SVN_URL="file:///tmp/wp-svn-actblue-contributions" ./.circleci/deploy-assets.sh -u actblue -p password
    SVN_URL="file:///tmp/wp-svn-actblue-contributions" ./.circleci/deploy-plugin.sh -u actblue -p password
    ```

5. Once complete, `cd` into the local svn directory. This is where we should see a representation of what we'd see on the WordPress svn server:

    ```
    cd /var/svn
    ```

6. Check this directory for our expected plugin files. The `/tmp/svn/trunk/` directory should have all of our distributed plugin files:

    <img width="704" alt="Screen Shot 2020-11-24 at 10 45 37 AM" src="https://user-images.githubusercontent.com/3286676/100138050-4c6f1680-2e42-11eb-91e4-08dba470b75d.png">

7. Check the `/tmp/svn/tags/` directory and confirm that you have a `1.0.0` directory in there. Then check that directory and confirm that it has all of the distributed plugin files (it should look the same as the `trunk/` directory).

    <img width="528" alt="Screen Shot 2020-11-24 at 10 47 26 AM" src="https://user-images.githubusercontent.com/3286676/100138232-92c47580-2e42-11eb-9ae2-150c835dd495.png">
    <img width="701" alt="Screen Shot 2020-11-24 at 10 47 38 AM" src="https://user-images.githubusercontent.com/3286676/100138236-93f5a280-2e42-11eb-9b84-92f0b4caae27.png">

## Todo

- [ ] Make sure this actually works when run with CircleCI.
